experiments/srupp_experiments/embedding.py

```diff
@@ -179,7 +179,7 @@ def _forward(self, hidden, target,
         if self.n_clusters == 0:
             return -head_logprob.gather(1, target.unsqueeze(1)).squeeze(1)
 
-        nll = torch.zeros_like(target, dtype=hidden.dtype, device=hidden.device)
+        nll = torch.zeros_like(target, dtype=head_logprob.dtype, device=head_logprob.device)
```

Comment:
might be worth adding a comment that nll is intended to  be fp32, and that head_logprob is fp32?

Reply:
(so we remember 12 months from now)

Reply:
i think it is okay. it is intended to be the same dtype as the tensor that will be copied from.